external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -138,10 +139,15 @@ class KafkaTestUtils extends Logging {
 
     if (server != null) {
       server.shutdown()
+      server.awaitShutdown()
       server = null
     }
 
-    brokerConf.logDirs.foreach { f => Utils.deleteRecursively(new File(f)) }
+    // On Windows, `logDirs` is left open even after Kafka server above is completely shut-downed
+    // in some cases. It leads to test failures on Windows if these are not ignored.
+    brokerConf.logDirs.map(new File(_))
+      .filterNot(FileUtils.deleteQuietly)
```

Comment:
I think I missed the point here, sorry: so, why do we think that FileUtils.deleteQuietly works but Utils.deleteRecursively does not? Is the point just that the former method is "quiet" instead of throwing an exception? I suppose I'd just try-catch around the Utils.deleteRecursively instead of use a different method in this case. Either way you have a path that logs a statement here so it's not so different either way.

I don't feel too strongly about it.